io/types.d.ts

```diff
@@ -1,57 +1,203 @@
+ *
+ * The format of each entry in emitBacks can be:
+ *
+ *   1. A single name string - the event name acts as the mutation
+ *   2. A string with a double-dashed arrow - the **right** side of the arrow is the event name, the **left** side is the mutation
+ *   3. An object - the object **key** is the mutation, the **value** is the event
+
+ */
+type EmitBackNotation = MutationNotation;
+
+
+/**
+ * Options to let you sync incoming events to a Vuex store and emit events
+ * on Vuex store changes. These options will override settings from your Nuxt
+ * configuration.
+ */
+interface NuxtSocketVueOptions {
+  mutations?: Array<MutationNotation>;
+  actions?: Array<ActionNotation>;
+  emitBacks?: Array<EmitBackNotation>;
+}
 
 interface NuxtSocketOpts extends SocketIOClient.ConnectOpts {
   /** Name of the socket. If omitted, the default socket will be used. */
   name?: string;
-  /** 
-   * The channel (a.k.a namespace) to connect to. 
+  /**
+   * The channel (a.k.a namespace) to connect to.
    * @default ''
    */
   channel?: string;
-  /** Specifies whether to enable or disable the "auto-teardown" feature 
-   * (see section below). 
-   * @default true 
+  /** Specifies whether to enable or disable the "auto-teardown" feature
+   * (see section below).
+   * @default true
    */
   teardown?: boolean;
-  /** 
+  /**
    * Specifies whether to persist this socket so it can be reused
-   * (see [vuexModule](https://nuxt-socket-io.netlify.app/vuexModule)). 
+   * (see [vuexModule](https://nuxt-socket-io.netlify.app/vuexModule)).
    * @default false
    */
   persist?: boolean;
   /** Specifies the property in [this] component that will be used
-   *  to contain the socket status (referring to an object). 
-   * @default 'socketStatus' 
+   *  to contain the socket status (referring to an object).
+   * @default 'socketStatus'
    */
   statusProp?: string;
-  /** Specifies the timeout in milliseconds for an emit event, 
-   * after which waiting for the emit response will be canceled. 
-   * @default undefined 
+  /** Specifies the timeout in milliseconds for an emit event,
+   * after which waiting for the emit response will be canceled.
+   * @default undefined
    */
   emitTimeout?: number;
-  /** 
-   * Specifies the property in [this] component that will be used 
-   * to contain emit errors (see section below). 
+  /**
+   * Specifies the property in [this] component that will be used
+   * to contain emit errors (see section below).
    * (referring to this.emitErrors, an object of arrays)
-   * @default 'emitErrors' 
+   * @default 'emitErrors'
    */
   emitErrorsProp?: string;
   namespaceCfg?: {
     emitters?: Array<string | object>;
     listeners?: Array<string | object>;
   };
+  /**
+   * @default 'ioApi'
+   */
+  ioApiProp?: string;
+  /**
+   * @default 'ioData'
+   */
+  ioDataProp?: string;
+  /**
+   * @default []
+   */
+  apiIgnoreEvts?: Array<string>;
+  serverAPI?: {
```

Comment:
Can probably reuse this type. I refer to it as the KissAPI type, that are reused by both client and server.